devito/ir/iet/utils.py

```diff
@@ -92,31 +92,33 @@ def compose_nodes(nodes, retrieve=False):
         return body
 
 
-def derive_parameters(nodes, drop_locals=False):
+def derive_parameters(iet, drop_locals=False):
     """
     Derive all input parameters (function call arguments) from an IET
     by collecting all symbols not defined in the tree itself.
     """
     # Pick all free symbols and symbolic functions from the kernel
-    functions = FindSymbols('symbolics').visit(nodes)
-    free_symbols = FindSymbols('free-symbols').visit(nodes)
+    functions = FindSymbols('symbolics').visit(iet)
```

Comment:
wouldn't something better be `sym_funcs = Find...(`sym_funcs`)....
Although we search for symbolic functions...
seems awkward to use 'symbolics' as a search key and assigning the returned stuff to funtions..
I know it has nothing to do with this PR per se... just some nitpicking personal view....